end_to_end/end_to_end_suite_test.go

```diff
@@ -2242,6 +2242,42 @@ LANGUAGE plpgsql NO SQL;`)
 				"tplain": 1000, "ths": 1000, "tht": 1000})
 		})
 	})
+	Describe("Properly hanldes user defined type if different schema used", func() {
+		It("restores table with enum when exporting statistics", func() {
+			testhelper.AssertQueryRuns(backupConn, `create schema cdm;
+													create type cdm.status AS ENUM ('ACTIVATING', 'ACTIVATED', 'ERROR');
+													create cast (text as cdm.status) with inout as implicit;`)
+			testhelper.AssertQueryRuns(backupConn, `create table cdm.contact_status (
+															contact_id uuid not null,
+															status cdm.status not null
+														) distributed by (contact_id);
+													insert into cdm.contact_status 
+													select 
+													md5(random()::text)::uuid,
+													(array['ACTIVATING', 'ACTIVATED', 'ERROR'])[1 + x%3]
+													from generate_series(1,20000) x;
+													analyze cdm.contact_status;
+												`)
+
+			defer func() {
+				testhelper.AssertQueryRuns(backupConn, `drop table cdm.contact_status;
+														drop cast (text as cdm.status);
+														drop type cdm.status cascade;
+														drop schema cdm;`)
+			}()
+
+			output := gpbackup(gpbackupPath, backupHelperPath,
+				"--backup-dir", backupDir,
+				"--with-stats")
+			timestamp := getBackupTimestamp(string(output))
+
+			gprestore(gprestorePath, restoreHelperPath, timestamp,
+				"--redirect-db", "restoredb",
+				"--backup-dir", backupDir)
+			assertDataRestored(restoreConn, map[string]int{
+				"cdm.contact_status": 20000})
+		})
```

Comment:
Why does the test pass without a patch?

Reply:
Also, symbols in the name break the behavior:
```diff
diff --git a/end_to_end/end_to_end_suite_test.go b/end_to_end/end_to_end_suite_test.go
index 134ab746..2614ad46 100644
--- a/end_to_end/end_to_end_suite_test.go
+++ b/end_to_end/end_to_end_suite_test.go
@@ -2245,11 +2245,11 @@ LANGUAGE plpgsql NO SQL;`)
 	Describe("Properly hanldes user defined type if different schema used", func() {
 		It("restores table with enum when exporting statistics", func() {
 			testhelper.AssertQueryRuns(backupConn, `create schema cdm;
-													create type cdm.status AS ENUM ('ACTIVATING', 'ACTIVATED', 'ERROR');
-													create cast (text as cdm.status) with inout as implicit;`)
+													create type cdm."status'" AS ENUM ('ACTIVATING', 'ACTIVATED', 'ERROR');
+													create cast (text as cdm."status'") with inout as implicit;`)
 			testhelper.AssertQueryRuns(backupConn, `create table cdm.contact_status (
 															contact_id uuid not null,
-															status cdm.status not null
+															status cdm."status'" not null
 														) distributed by (contact_id);
 													insert into cdm.contact_status 
 													select 
@@ -2261,8 +2261,8 @@ LANGUAGE plpgsql NO SQL;`)
 
 			defer func() {
 				testhelper.AssertQueryRuns(backupConn, `drop table cdm.contact_status;
-														drop cast (text as cdm.status);
-														drop type cdm.status cascade;
+														drop cast (text as cdm."status'");
+														drop type cdm."status'" cascade;
 														drop schema cdm;`)
 			}()
 
@@ -2273,7 +2273,8 @@ LANGUAGE plpgsql NO SQL;`)
 
 			gprestore(gprestorePath, restoreHelperPath, timestamp,
 				"--redirect-db", "restoredb",
-				"--backup-dir", backupDir)
+				"--backup-dir", backupDir,
+				"--with-stats")
 			assertDataRestored(restoreConn, map[string]int{
 				"cdm.contact_status": 20000})
 		})
```
```
20240917:08:59:20 gprestore:gpadmin:gpdb6:047119-[DEBUG]:-Error encountered when executing statement: INSERT INTO pg_statistic SELECT
	attrelid,
	attnum,
	false::boolean,
	0.000000::real,
	4::integer,
	3.000000::real,
	1::smallint,
	3::smallint,
	0::smallint,
	0::smallint,
	0::smallint,
	3516::oid,
	3518::oid,
	0::oid,
	0::oid,
	0::oid,
	'{"0.333350003","0.333350003","0.333299994"}'::real[],
	'{"0.329271734"}'::real[],
	NULL::real[],
	NULL::real[],
	NULL::real[],
	array_in('{"ACTIVATED","ERROR","ACTIVATING"}', 'cdm."status'"'::regtype::oid, -1),
	NULL,
	NULL,
	NULL,
	NULL
FROM pg_attribute WHERE attrelid = 'cdm.contact_status'::regclass::oid AND attname = 'status'; Error was: ERROR: unterminated quoted identifier at or near ""'::regtype::oid, -1),
	NULL,
	NULL,
	NULL,
	NULL
FROM pg_attribute WHERE attrelid = 'cdm.contact_status'::regclass::oid AND attname = 'status';
" (SQLSTATE 42601)

20240917:08:59:20 gprestore:gpadmin:gpdb6:047119-[CRITICAL]:-ERROR: unterminated quoted identifier at or near ""'::regtype::oid, -1),
	NULL,
	NULL,
	NULL,
	NULL
FROM pg_attribute WHERE attrelid = 'cdm.contact_status'::regclass::oid AND attname = 'status';
" (SQLSTATE 42601)
```

Reply:
I confirm it fails in this case, but it's in the scope of different issue, not this one.

Reply:
> Why does the test pass without the patch? I think you forgot to add "--with-stats" when restoring!

Right, added it with extra check.